talisker/postgresql.py

```diff
@@ -0,0 +1,142 @@
+# Copyright (C) 2016- Canonical Ltd
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+from __future__ import unicode_literals
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+
+from builtins import *  # noqa
+
+import collections
+import logging
+import math
+import time
+import shlex
+
+from psycopg2.extensions import cursor, connection
+
+import sqlparse
+import raven.breadcrumbs
+import talisker
+
+__all__ = [
+    'TaliskerConnection',
+    'TaliskerCursor',
+    'prettify_sql',
+]
+
+
+FILTERED = '<query filtered>'
+
+
+def prettify_sql(sql):
+    if sql is None:
+        return None
+    return sqlparse.format(
+        sql,
+        keyword_case="upper",
+        identfier_case="lower",
+        strip_comments=False,
+        reindent=True,
+        indent_tabs=False)
+
+
+def get_safe_connection_string(conn):
+    try:
+        # 2.7+
+        params = conn.get_dsn_parameters()
+    except AttributeError:
+        params = dict(i.split('=') for i in shlex.split(conn.dsn))
+    return '{user}@{host}:{port}/{dbname}'.format(**params)
+
+
+class TaliskerConnection(connection):
+    _logger = None
+    _mintime = None
+
+    @property
+    def logger(self):
+        if self._logger is None:
+            self._logger = logging.getLogger('talisker.slowqueries')
+        return self._logger
+
+    @property
+    def query_mintime(self):
+        if self._mintime is None:
+            self._mintime = talisker.get_config()['slowquery_time']
+        return self._mintime
+
+    def cursor(self, *args, **kwargs):
+        kwargs.setdefault('cursor_factory', TaliskerCursor)
+        return super().cursor(*args, **kwargs)
```

Comment:
You can with python-future.org :) It's a bit lazy of me, as being explicit is supported in both, but I am so. very. tired. of explicit super in py2, so I just go all out and rely future's py2 workaround.